lib/resty/etcdv3.lua

```diff
@@ -353,34 +353,60 @@ end
 
 do
 
-function _M.get(self, key)
+function _M.get(self, key, timeout, revision)
     if not typeof.string(key) then
         return nil, 'key must be string'
     end
 
+    attr.timeout  = timeout
+    attr.revision = revision
+
     return get(self, key)
 end
 
     local attr = {}
-function _M.watch(self, key, timeout)
+function _M.watch(self, key, timeout, start_revision, progress_notify, filters, prev_kv, watch_id, fragment)
```

Comment:
how about a lua table to store the options?

`_M.watch(self, key, timeout, opts)`

```lua
local opts = {
    start_revision = 1, 
    progress_notify = 2,
    ....
}
```